compiler/ccgstmts.nim

```diff
@@ -490,8 +490,12 @@ iterator fieldValuePairs(n: PNode): tuple[memberSym, valueSym: PNode] =
         let memberSym = identDefs[i]
         yield((memberSym: memberSym, valueSym: valueSym))
 
-proc genComputedGoto(p: BProc; n: PNode) =
+proc genComputedGoto(p: BProc; nAsIs: PNode) =
   # first pass: Generate array of computed labels:
+
+  # flatten the loop body because otherwise let and var sections
+  # wrapped inside stmt lists by inject destructors won't be recognised
+  let n = nAsIs.flattenStmts()
```

Comment:
Btw, you can do `let n = n.flattenStmts()` too, it will just shadow the argument n.

Reply:
Oh right, I totally forgot about shadowing!